app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ActiveRecord::Base
+  belongs_to :user
+  has_many :comments
+  has_many :likes
+
+  def update_posts_counter
```

Comment:
[ OPTIONAL ] It's a suggestion that you can use Active Record Call back `after_save` to call the `#update_posts_counter` method every time a new post has been saved to database, and make the `update_posts_counter` method private.